core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.directive.ts

```diff
@@ -72,7 +72,7 @@ angular.module('oppia').directive('topNavigationBar', [
                     $window.location = loginUrl;
                   }, 150);
                 } else {
-                  throw Error('Login url not found.');
+                  $window.location.reload();
```

Comment:
What if the `loginUrl` is not defined for some different reason other than the user is already logged in? Should we still reload? Ditto for the other files.

Reply:
From what I understand, it looks like the only condition where login_url is None would be when self.user_id is None. I think we use this to check if the user has logged in in most places. See code from profile.py below.

```py
@acl_decorators.open_access
    def get(self):
        login_url = None
        if self.user_id:
            self.render_json({'login_url': None})
        else:
            if self.request and self.request.get('current_url'):
                target_url = (
                    '/' if self.request.get('current_url').endswith(
                        feconf.SPLASH_URL)
                    else self.request.get('current_url'))
                login_url = (
                    current_user_services.create_login_url(target_url))
                self.render_json({'login_url': login_url})
            else:
                raise self.InvalidInputException(
                    'Incomplete or empty GET parameters passed'
                )
```

Reply:
Okay

Reply:
Although, I wonder if `current_user_services.create_login_url(target_url)` can return None in some case. Do you think we should determine this and log it somehow so that it doesn't slip past us? (considering that I'm removing the error in the frontend.)

Reply:
Maybe check the code for `create_login_url` to see if it is possible that it returns `None`.

Reply:
Looks like it uses the GAE users api method create_login_url. According to [this page](https://cloud.google.com/appengine/docs/standard/python/refdocs/google.appengine.api.users) and source code [here](https://cloud.google.com/appengine/docs/standard/python/refdocs/modules/google/appengine/api/users#create_login_url), it looks like it will always return a string or raises an error. So I guess we should be fine.